packages/core/src/modules/non-secret/service/NonSecretRecordService.ts

```diff
@@ -0,0 +1,40 @@
+import type { ConnectionRecord } from '../../connections/repository/ConnectionRecord'
+import type { NonSecretRecordTags } from '../repository/NonSecretRecord'
+
+import { Lifecycle, scoped } from 'tsyringe'
+
+import { AriesFrameworkError } from '../../../error'
+import { NonSecretRecord } from '../repository/NonSecretRecord'
+import { NonSecretRecordRepository } from '../repository/NonSecretRecordRepository'
+
+@scoped(Lifecycle.ContainerScoped)
+export class NonSecretRecordService {
+  private nonSecretRepository: NonSecretRecordRepository
+
+  public constructor(nonSecretRepository: NonSecretRecordRepository) {
+    this.nonSecretRepository = nonSecretRepository
+  }
+
+  public async saveRecord(message: string, tags?: NonSecretRecordTags, connectionRecord?: ConnectionRecord) {
+    const nonsecretRecord = new NonSecretRecord({
+      content: message,
+      connectionId: connectionRecord?.id,
+      tags: tags,
+    })
+
+    try {
+      await this.nonSecretRepository.save(nonsecretRecord)
+      return nonsecretRecord
+    } catch (error) {
+      throw new AriesFrameworkError(`${error}`)
```

Comment:
This error should get some context

```suggestion
      throw new AriesFrameworkError(`Unable to store the nonSecret record with id ${nonSecretRecord.id}. Message: ${error}`)
```